app/libraries/AdminUtilities.php

```diff
@@ -28,7 +28,8 @@ public static function add_gateway($inputs)
         $gateway->gatewayAdminEmail = $inputs["admin-email"];
         $gateway->gatewayAdminLastName = $inputs["admin-lastname"];
         $gateway->identityServerUserName = $inputs["admin-username"];
-        $gateway->identityServerPasswordToken  = $inputs["admin-password"];
+        $token = create_pwd_token([Session::get('username'), $inputs["admin-password"]]);
```

Comment:
I think you have to call create_pwd_token like `AdminUtilities::create_pwd_token`. Also the array passed should specify the keys.  Something like
```
AdminUtilities::create_pwd_token([
  "username" => $inputs["admin-username"], 
  "password" => $inputs["admin-password"],
  "description" => "Admin user password for gateway " . $id
]);